install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,492 @@ toc::[]
+the service. This translates into a final backend implementation for routers.
+|===
+ServiceUnit is an encapsulation of a service, the endpoints that back
+that service, and the routes that point to the service.  This is the
+data that drives the creation of the router configuration files
+
+
+.Router Endpoint
+[cols="2*", options="header"]
+|===
+|Field | Type
+|`*ID*` | string
+|`*IP*` | string
+|`*Port*` | string
+|`*TargetName*` | string
+|`*PortName*` | string
+|`*IdHash*` | string
+|`*NoHealthCheck*` | bool
+|===
+Endpoint is an internal representation of a Kubernetes endpoint.
+
+.Router Certificate, ServiceAliasConfigStatus
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|`*Certificate*` | string | Represents a pub/private key pair.  It is
+identified by ID which will become the file name. A CA certificate will
+not have a PrivateKey set.
+|`*ServiceAliasConfigStatus*` | string | Indicates that the necessary
+files for this configuration have been persisted to disk. Valid values "saved", "".
+|===
+
+.Router Certificate Type
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|ID | string|
+|Contents| string| The certificate.
+|PrivateKey|string| The private key.
+|===
+
+.Router TLSTerminationType
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|`*TLSTerminationType*` | string | TLSTerminationType dictate where the
+secure communication will stop.
+|`*InsecureEdgeTerminationPolicyType*` | string | insecureEdgeTerminationPolicy
+indicates the desired behavior for insecure connections to a route. While
+each router may make its own decisions on which ports to expose, this is
+normally port 80.
+|===
+
+TLSTerminationType and InsecureEdgeTerminationPolicyType dictate where
+the secure communication will stop.
+
+
+.Router TLSTerminationType Values
+[cols="3*", options="header"]
+|===
+|Constant | Value | Meaning
+|TLSTerminationEdge | `*edge*` | Terminate encryption at the edge router.
+|TLSTerminationPassthrough | `*passthrough*` | Terminate encryption at
+the destination, the destination is responsible for decrypting traffic.
+|TLSTerminationReencrypt | `*reencrypt*` | Terminate encryption at the edge
+router and re-encrypt it with a new certificate supplied by the destination.
+|===
+
+.Router InsecureEdgeTerminationPolicyType Values
+[cols="2*", options="header"]
+|===
+|Type | Meaning
+|`*Allow*` | traffic is sent to the server on the insecure port (default)
+|`*Disable*` | no traffic is allowed on the insecure port.
+|`*Redirect*` | clients are redirected to the secure port.
+|===
+None ("") is the same as `*Disable*`.
+
+[[using-annotations]]
+=== Annotations
+
+Each route can have annotations attached. Each annotation is just a name
+and a value.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/timeout: 5500ms
+[...]
+----
+
+The name can be pretty much anything that doesn't conflict with existing
```

Comment:
Should remove "pretty much." Do we want to say "The name can be anything..." or are there any restrictions we should mention? Or, just "Make sure the name doesn't conflict with existing Annorations"